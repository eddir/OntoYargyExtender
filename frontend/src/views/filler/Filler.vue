<template>
  <div>
    <CRow>
      <CCol ref="filler" md="12">
        <NewOntologyFill @update="start"></NewOntologyFill>
      </CCol>
      <CCol ref="wip" md="12" hidden>
        <CContainer>
          <CRow>
            <CCol md="12">
              <CCard>
                <CCardHeader>
                  Заполнение онтологий
                </CCardHeader>
                <CCardBody>
                  <CSpinner
                      color="primary"
                      style="width:4rem;height:4rem;"
                  />
                </CCardBody>
              </CCard>
            </CCol>
          </CRow>
        </CContainer>
      </CCol>
    </CRow>
  </div>
</template>
<script>
import NewOntologyFill from "@/views/ontologies/NewOntologyFill.vue";
import API from "@/services/API.vue";

export default {
  name: 'Filler',
  components: {NewOntologyFill},
  data() {
    return {
      channel: null,
    }
  },
  created() {
    this.channel = this.$pusher.subscribe('ontologies-tasks');
    this.channel.bind('fill-event', data => {
      this.finish(data['ontology_id']);
    });
  },
  beforeDestroy() {
    this.channel.unbind();
    this.$pusher.unsubscribe('ontologies-tasks');
  },
  methods: {
    start() {
      this.$refs.filler.hidden = true;
      this.$refs.wip.hidden = false;
    },
    finish(ontology_id) {
      this.$refs.filler.hidden = false;
      this.$refs.wip.hidden = true;
      this.$toast.success("Онтология заполнена");
      API.downloadFilledOntology(ontology_id)
    },
  }
}
</script>
