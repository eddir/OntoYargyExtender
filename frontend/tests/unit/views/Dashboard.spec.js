import Vue from 'vue'
import regeneratorRuntime from "regenerator-runtime";
import { shallowMount } from '@vue/test-utils'
import CoreuiVue from '@coreui/vue'
import Filler from '@/views/filler/Filler.vue'


Vue.use(regeneratorRuntime)
Vue.use(CoreuiVue)

describe('Filler.vue', () => {
  it('has a name', () => {
    expect(Filler.name).toBe('Filler')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Filler)
    expect(wrapper.vm).toBeTruthy()
  })
  it('is Dashboard', () => {
    const wrapper = shallowMount(Filler)
    expect(wrapper.findComponent(Filler)).toBeTruthy()
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Filler)
    expect(wrapper.element).toMatchSnapshot()
  })
})
