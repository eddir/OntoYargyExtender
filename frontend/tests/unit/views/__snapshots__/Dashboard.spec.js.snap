// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard.vue renders correctly 1`] = `
<div>
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      md="12"
      tag="div"
    >
      <ontologies-stub
        onto_list=""
      />
       
      <newontologyfill-stub />
    </ccol-stub>
  </crow-stub>
</div>
`;
